jlm/tooling/CommandLine.hpp

```diff
@@ -54,6 +54,7 @@ class JlmOptCommandLineOptions final : public CommandLineOptions
     Ascii,
     Llvm,
     Mlir,
+    Tree,
```

Comment:
Adding something into this class will always require changes to the unordered_map in the cpp (line 311 above).
Would it make sense to declare these in the same place and then use them where they are needed?

This resembles the LLVM attributes that need to be updated when we change LLVM version, which requires 3 different files to be updated.

Reply:
Yeah, you are right about it. The problem originates from us requiring a string version of the output format value, but c++ does not offer a "ToString()" method on enum types by default (which is insane in 2024). This means we need to implement this ourselves (with this unordered_map).

We have a unit test that ensures we are also adding a line in the map if we add a new output format. The unit test fails if you only add a new output format, but forget the line in the map. The best we can do in the absence of language/compiler support.